File: src/ast/NameTypeWithTemplateArguments.java
package ast;

import java.io.StringWriter;

public class NameTypeWithTemplateArguments extends BaseNode {
    private BaseNode _prev;
    private BaseNode _templateArgument;

    public NameTypeWithTemplateArguments(BaseNode prev, BaseNode templateArgument) {
        super(NodeType.NameTypeWithTemplateArguments);
        _prev = prev;
        _templateArgument = templateArgument;
    }

    @Override
    public String GetName() {
        return _prev.GetName();
    }

    @Override
    public void PrintLeft(StringWriter writer) {
        _prev.Print(writer);
        _templateArgument.Print(writer);
    }
}

File: src/de/fearlesstobi/demangler/ast/NameTypeWithTemplateArguments.kt
package de.fearlesstobi.demangler.ast

import java.io.StringWriter

class NameTypeWithTemplateArguments(private val prev: BaseNode, private val templateArgument: BaseNode) : BaseNode(NodeType.NameTypeWithTemplateArguments) {
    override fun getName(): String? {
        return prev.getName()
    }

    public override fun printLeft(writer: StringWriter) {
        prev.print(writer)
        templateArgument.print(writer)
    }

}